closed/adds/jdk/src/share/classes/jdk/crypto/jniprovider/NativeCrypto.java

```diff
@@ -86,6 +86,17 @@ public class NativeCrypto {
             return osslVersion;
         }).longValue();
 
+    private static final boolean isOpenSSLFIPS = AccessController.doPrivileged(
+        (PrivilegedAction<Boolean>) () -> {
+        boolean isFIPS;
+        if (ossl_ver != -1) {
+            isFIPS = isOpenSSLFIPS();
+        } else {
+            isFIPS = false;
+        }
+        return isFIPS;
```

Comment:
This doesn't require privileges; it can be simplified to:
```java
    private static final boolean isOpenSSLFIPS = (ossl_ver != -1) && isOpenSSLFIPS();
```

Reply:
Yes, you're right. Changed to suggestion.